host_core/lib/host_core/jetstream/client.ex

```diff
@@ -123,85 +124,263 @@ defmodule HostCore.Jetstream.Client do
         )
     end
 
-    {:noreply, state}
+    {:noreply, state, {:continue, :create_legacy_eph_consumer}}
   end
 
-  def handle_consumer_create_response(%{
-        "type" => "io.nats.jetstream.api.v1.consumer_create_response",
-        "error" => %{
-          "code" => code,
-          "description" => desc
+  # Creates an ephemeral consumer against the legacy LATTICECACHE_{prefix} if such a stream exists. This will
+  # migrate all the data from that stream into the new metadata cache and then delete it.
+  @impl true
+  def handle_continue(:create_legacy_eph_consumer, state) do
+    stream_name = "LATTICECACHE_#{state.lattice_prefix}"
+
+    if stream_exists?(state.lattice_prefix, stream_name, state.domain) do
+      Logger.warn("Detected deprecated lattice cache stream #{stream_name}.")
+
+      Logger.warn(
+        "The LATTICECACHE_#{state.lattice_prefix} stream will be removed. You should not start actors, providers, or add metadata from older hosts relying on that stream."
+      )
```

Comment:
I do get this message for a stream that I created with an old host, but link definitions don't seem to actually migrate from that stream to the new kv bucket (and if I start an old host, I get the link definitions back, so the stream isn't being deleted either)

Reply:
🤔 this is weird, as I've actually watched the old stream go away multiple times